hoard/sources/whoas.py

```diff
@@ -47,22 +47,26 @@ def create_from_whoas_dim_xml(data: str) -> Dataset:
     kwargs: Dict[str, Any] = {}
     record = ET.fromstring(data)
     fields = record.findall(".//dim:field", namespace)
-    authors = []
-    contacts = [
+    kwargs["contacts"] = [
         Contact(
             datasetContactName="NAME, FAKE",
             datasetContactEmail="FAKE_EMAIL@EXAMPLE.COM",
         )
     ]
-    descriptions = []
-    distributors = []
-    grantNumbers = []
-    keywords = []
-    languages = []
-    notesText = None
-    otherIds = []
-    publications = []
-    timePeriodsCovered = []
+    list_keys = [
+        "authors",
+        "description",
+        "keywords",
+        "distributors",
+        "grantNumbers",
+        "language",
+        "otherIds",
+        "publications",
+        "timePeriodsCovered",
+    ]
+    for key in list_keys:
+        kwargs.setdefault(key, [])
```

Comment:
Sorry, what I meant was we should use `setdefault` when we actually try to append to a field. For example:
```python
kwargs.setdefault("authors", []).append(Author...)
```

Reply:
Got it, this is my first time with `setdefault`, I'll fix and this is a good pattern to know going forward. Thanks!

Reply:
Fixed